tests/parametric_study/test_parametric_study.py

```diff
@@ -1736,3 +1737,163 @@ def test_set_iteration_sets_iteration(tmp_path: pytest.TempPathFactory):
+        study.import_csv_study(filename)
+
+
+def test_import_csv_study_raises_exception_when_file_is_not_a_csv(tmp_path: pytest.TempPathFactory):
+    # arrange
+    study_name = "test_study"
+    filename = __file__
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    with pytest.raises(ValueError, match="does not have the expected columns"):
+        study.import_csv_study(filename)
+
+
+def test_import_csv_study_raises_exception_when_file_does_not_have_correct_headers(
+    tmp_path: pytest.TempPathFactory,
+):
+    # arrange
+    study_name = "test_study"
+    incorrect_headers_file = test_utils.get_test_file_path(
+        pathlib.Path("csv") / "incorrect-headers.csv"
+    )
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    with pytest.raises(ValueError, match="does not have the expected columns"):
+        study.import_csv_study(incorrect_headers_file)
+
+
+def test_import_csv_study_adds_simulations_to_new_study(tmp_path: pytest.TempPathFactory):
+    # arrange
+    study_name = "test_study"
+    single_bead_demo_csv_file = test_utils.get_test_file_path(
+        pathlib.Path("csv") / "single-bead-test-study.csv"
+    )
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    study.import_csv_study(single_bead_demo_csv_file)
+    assert len(study.data_frame()) == 5
+    assert len(study.data_frame()[ps.ColumnNames.LASER_POWER].unique()) == 5
+
+
+def test_import_csv_study_adds_simulations_of_multiple_input_types_to_new_study(
+    tmp_path: pytest.TempPathFactory,
+):
+    # arrange
+    study_name = "test_study"
+    single_bead_demo_csv_file = test_utils.get_test_file_path(
+        pathlib.Path("csv") / "single-bead-test-study.csv"
+    )
+    porosity_demo_csv_file = test_utils.get_test_file_path(
+        pathlib.Path("csv") / "porosity-test-study.csv"
+    )
+    microstructure_demo_csv_file = test_utils.get_test_file_path(
+        pathlib.Path("csv") / "microstructure-test-study.csv"
+    )
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    study.import_csv_study(single_bead_demo_csv_file)
+    study.import_csv_study(porosity_demo_csv_file)
+    study.import_csv_study(microstructure_demo_csv_file)
+    assert len(study.data_frame()) == 15
+
+
+def test_import_csv_study_adds_simulations_to_existing_study(tmp_path: pytest.TempPathFactory):
+    # arrange
+    study_name = "test_study"
+    single_bead_demo_csv_file = test_utils.get_test_file_path(
+        pathlib.Path("csv") / "single-bead-test-study.csv"
+    )
+    # act, assert
+    study = ps.ParametricStudy(tmp_path / study_name)
+    powers = [50, 250, 700]
+    scan_speeds = [0.35]
+
+    # act
+    study.generate_porosity_permutations(
+        material_name="material",
+        laser_powers=powers,
+        scan_speeds=scan_speeds,
+    )
+
+    # act
+    study.import_csv_study(single_bead_demo_csv_file)
+    assert len(study.data_frame()) == 8
```

Comment:
The general convention is to have one `# act` statement and one `# assert` section. Here `import_csv_study()` is the function under test so our "action" is to call the function after setting up the preconditions. All the preconditions, which may include calling multiple functions, are part of the `# arrange` section.

```suggestion
    # arrange
    study_name = "test_study"
    single_bead_demo_csv_file = test_utils.get_test_file_path(
        pathlib.Path("csv") / "single-bead-test-study.csv"
    )
    study = ps.ParametricStudy(tmp_path / study_name)
    powers = [50, 250, 700]
    scan_speeds = [0.35]
    study.generate_porosity_permutations(
        material_name="material",
        laser_powers=powers,
        scan_speeds=scan_speeds,
    )

    # act
    study.import_csv_study(single_bead_demo_csv_file)
    
    # assert
    assert len(study.data_frame()) == 8
```